src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -45,6 +48,43 @@ def _set_uniq_aggregation_on_session_status_factory(
     )
 
 
+def _aggregation_on_tx_status_func_factory(aggregate):
+    def _snql_on_tx_status_factory(org_id, exclude_tx_statuses: List[str], metric_ids, alias=None):
+        resolved_tx_statuses = [resolve_weak(s) for s in exclude_tx_statuses]
+        return Function(
+            aggregate,
+            [
+                Column("value"),
+                Function(
+                    "and",
+                    [
+                        Function(
+                            "notIn",
+                            [
+                                Column(
+                                    f"tags[{resolve_weak(org_id, TransactionMetricsKey.TRANSACTION_STATUS.value)}]"
+                                ),
+                                resolved_tx_statuses,
+                            ],
+                        ),
+                        Function("in", [Column("metric_id"), list(metric_ids)]),
+                    ],
+                ),
+            ],
+            alias,
+        )
+
+    return _snql_on_tx_status_factory
+
+
+def _dist_sum_aggregation_on_tx_status_factory(
+    org_id, exclude_tx_statuses: List[str], metric_ids, alias=None
```

Comment:
Generalizing the function to accept `exclude_tx_statuses` isn't useful at this point, but this generalization is required in future metrics.